cirq-core/cirq/ops/pauli_string.py

```diff
@@ -698,6 +711,8 @@ def __pow__(self, power):
             return PauliString(
                 qubit_pauli_map=self._qubit_pauli_map, coefficient=self.coefficient**-1
             )
+        if self._is_parameterized_():
+            raise ValueError('Cannot raise to power when parameterized.')
```

Comment:
Should we just return `NotImplemented`, like done below for non-unitary PauliStrings?